Cannot succeed to install

I have installed libfreenect2 in my home directory.
This create freenect2 in my home directory too, containing /include and /lib
Protonect, NiViewer2 and Nite2 are working.
Try installing pylibfreenect2 with:
sudo LIBFREENECT2_INSTALL_PREFIX=/home/stf/freenect2/ pip install git+https://github.com/r9y9/pylibfreenect2
getting errors:

Downloading/unpacking git+https://github.com/r9y9/pylibfreenect2
  Cloning https://github.com/r9y9/pylibfreenect2 to /tmp/pip-3jVmAQ-build
  Running setup.py (path:/tmp/pip-3jVmAQ-build/setup.py) egg_info for package from git+https://github.com/r9y9/pylibfreenect2
    Compiling pylibfreenect2/libfreenect2.pyx because it changed.
    Cythonizing pylibfreenect2/libfreenect2.pyx
    
Requirement already satisfied (use --upgrade to upgrade): numpy>=1.7.0 in /usr/local/lib/python2.7/dist-packages (from pylibfreenect2==0.0.1-dev)
Requirement already satisfied (use --upgrade to upgrade): enum34 in /usr/local/lib/python2.7/dist-packages (from pylibfreenect2==0.0.1-dev)
Installing collected packages: pylibfreenect2
  Running setup.py install for pylibfreenect2
    building 'pylibfreenect2.libfreenect2' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/home/stf/freenect2/include/libfreenect2 -I/usr/include/python2.7 -c pylibfreenect2/libfreenect2.cpp -o build/temp.linux-x86_64-2.7/pylibfreenect2/libfreenect2.o -std=c++11
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
    In file included from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1781:0,
                     from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                     from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
                     from pylibfreenect2/libfreenect2.cpp:287:
    /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
     #warning "Using deprecated NumPy API, disable it by " \
      ^
    In file included from pylibfreenect2/libfreenect2.cpp:298:0:
    /home/stf/freenect2/include/libfreenect2/frame_listener_impl.h:35:43: fatal error: libfreenect2/frame_listener.hpp: Aucun fichier ou dossier de ce type
     #include <libfreenect2/frame_listener.hpp>
                                               ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-3jVmAQ-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Om021S-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/pylibfreenect2

copying pylibfreenect2/__init__.py -> build/lib.linux-x86_64-2.7/pylibfreenect2

running build_ext

building 'pylibfreenect2.libfreenect2' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/pylibfreenect2

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/local/lib/python2.7/dist-packages/numpy/core/include -I/home/stf/freenect2/include/libfreenect2 -I/usr/include/python2.7 -c pylibfreenect2/libfreenect2.cpp -o build/temp.linux-x86_64-2.7/pylibfreenect2/libfreenect2.o -std=c++11

cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]

In file included from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1781:0,

                 from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/ndarrayobject.h:18,

                 from /usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/arrayobject.h:4,

                 from pylibfreenect2/libfreenect2.cpp:287:

/usr/local/lib/python2.7/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]

 #warning "Using deprecated NumPy API, disable it by " \

  ^

In file included from pylibfreenect2/libfreenect2.cpp:298:0:

/home/stf/freenect2/include/libfreenect2/frame_listener_impl.h:35:43: fatal error: libfreenect2/frame_listener.hpp: Aucun fichier ou dossier de ce type

 #include <libfreenect2/frame_listener.hpp>

                                           ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip-3jVmAQ-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Om021S-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip-3jVmAQ-build
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==1.5.4', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 235, in main
    return command.main(cmd_args)
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 42: ordinal not in range(128)

> /home/stf/freenect2/include/libfreenect2/frame_listener_impl.h:35:43: fatal error: libfreenect2/frame_listener.hpp: Aucun fichier ou dossier de ce type

I can reproduce the problem; libfreenect2 headers wasn't in header search paths with non-default libfreenect2 installation path. #10 should fix this. I will merge #10 when it's green.

For the record, the exact error log I saw was:

```
% LIBFREENECT2_INSTALL_PREFIX=~/freenect2 python setup.py develop   
/Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/setuptools-20.1.1-py3.5.egg/setuptools/dist.py:285: UserWarning: Normalizing '0.0.1-dev' to '0.0.1.dev0'
running develop
running egg_info
writing requirements to pylibfreenect2.egg-info/requires.txt
writing dependency_links to pylibfreenect2.egg-info/dependency_links.txt
writing top-level names to pylibfreenect2.egg-info/top_level.txt
writing pylibfreenect2.egg-info/PKG-INFO
reading manifest file 'pylibfreenect2.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pylibfreenect2.egg-info/SOURCES.txt'
running build_ext
building 'pylibfreenect2.libfreenect2' extension
gcc -fno-strict-aliasing -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/ryuyamamoto/anaconda/envs/py35/include -arch x86_64 -I/Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/numpy/core/include -I/Users/ryuyamamoto/freenect2/include -I/Users/ryuyamamoto/a
naconda/envs/py35/include/python3.5m -c pylibfreenect2/libfreenect2.cpp -o build/temp.macosx-10.5-x86_64-3.5/pylibfreenect2/libfreenect2.o -std=c++11 -stdlib=libc++ -mmacosx-version-min=10.8
In file included from pylibfreenect2/libfreenect2.cpp:283:
In file included from /Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
In file included from /Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/numpy/core/include/numpy/ndarraytypes.h:1781:
/Users/ryuyamamoto/anaconda/envs/py35/lib/python3.5/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by " \
 ^
pylibfreenect2/libfreenect2.cpp:293:10: fatal error: 'frame_listener.hpp' file not found
#include "frame_listener.hpp"
         ^
1 warning and 1 error generated.
```

--- a/pylibfreenect2/libfreenect2/libfreenect2.pxd
+++ b/pylibfreenect2/libfreenect2/libfreenect2.pxd
@@ -10,7 +10,7 @@ from libcpp.string cimport string
 from libcpp.map cimport map
 
 
-cdef extern from "frame_listener.hpp" namespace "libfreenect2":
+cdef extern from "libfreenect2/frame_listener.hpp" namespace "libfreenect2":
     # ugly but works
     cdef enum LibFreenect2FrameType "libfreenect2::Frame::Type":
         Color "libfreenect2::Frame::Type::Color"
@@ -33,7 +33,7 @@ cdef extern from "frame_listener.hpp" namespace "libfreenect2":
     cdef cppclass FrameListener:
         bool onNewFrame(int, Frame*)
 
-cdef extern from "frame_listener_impl.h" namespace "libfreenect2":
+cdef extern from "libfreenect2/frame_listener_impl.h" namespace "libfreenect2":
     cdef cppclass SyncMultiFrameListener:
         SyncMultiFrameListener(unsigned int)
 
@@ -41,7 +41,7 @@ cdef extern from "frame_listener_impl.h" namespace "libfreenect2":
         void waitForNewFrame(map[LibFreenect2FrameType, Frame*]&)
         void release(map[LibFreenect2FrameType, Frame*]&)
 
-cdef extern from "libfreenect2.hpp" namespace "libfreenect2":
+cdef extern from "libfreenect2/libfreenect2.hpp" namespace "libfreenect2":
     cdef cppclass Freenect2Device:
         unsigned int VendorId
         unsigned int ProductId
@@ -93,14 +93,14 @@ cdef extern from "libfreenect2.hpp" namespace "libfreenect2":
         void stop()
         void close()
 
-cdef extern from "registration.h" namespace "libfreenect2":
+cdef extern from "libfreenect2/registration.h" namespace "libfreenect2":
     cdef cppclass Registration:
         Registration(Freenect2Device.IrCameraParams, Freenect2Device.ColorCameraParams) except +
 
         # undistort/register a whole image
         void apply(const Frame*, const Frame*, Frame*, Frame*, const bool, Frame*, int*) const
 
-cdef extern from "packet_pipeline.h" namespace "libfreenect2":
+cdef extern from "libfreenect2/packet_pipeline.h" namespace "libfreenect2":
     cdef cppclass PacketPipeline:
         PacketPipeline *getRgbPacketParser() const
         PacketPipeline *getIrPacketParser() const
@@ -115,7 +115,7 @@ cdef extern from "packet_pipeline.h" namespace "libfreenect2":
         OpenCLPacketPipeline(const int)
 
 
-cdef extern from "libfreenect2.hpp" namespace "libfreenect2":
+cdef extern from "libfreenect2/libfreenect2.hpp" namespace "libfreenect2":
     cdef cppclass Freenect2:
         Freenect2() except +
 

--- a/setup.py
+++ b/setup.py
@@ -81,7 +81,7 @@ ext_modules = cythonize(
             join("pylibfreenect2", "libfreenect2" + ext),
         ],
         include_dirs=[np.get_include(),
-                      join(libfreenect2_include_top, "libfreenect2")],
+                      join(libfreenect2_include_top)],
         library_dirs=[libfreenect2_library_path],
         libraries=["freenect2"],
         extra_compile_args=extra_compile_args,

